### PR TITLE
Fixed In ContainerServiceManager, on KubernetesClusters creation, network profile definition: add dataplane and plugin mode

### PR DESCRIPTION
**Pull request by v-hongli1**
relate to [issue#39427](https://github.com/Azure/azure-sdk-for-java/issues/39427)

1. Add `withNetworkPluginMode, withNetworkMode, withNetworkDataPlan` interfaces in `Create` mode.
3. The `NetworkPlugin, NetworkPluginMode, NetworkMode` does not support update operation.

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java [hunk 1]
```diff
@@ -707,6 +759,22 @@ interface WithNetworkProfile {
...
+             */
+            Update withNetworkPolicy(NetworkPolicy networkPolicy);
+
+            /**
+             * Updates the cluster's network data plam.
+             *
+             * @param networkDataPlan the cluster's networkDataPlan
+             * @return the next stage of the update
+             */
+            Update withNetworkDataPlan(NetworkDataplane networkDataPlan);
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java:777]: Sorry, my fault.

We already had `withNetworkProfile(ContainerServiceNetworkProfile networkProfile);`. Please remove this 2 (and the impl in KubernetesClusterImpl.java)

Please keep the test (use `withNetworkProfile`) to set them (I didn't see update on `networkMode`, it there a reason?).

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java [hunk 2]
```diff
@@ -447,6 +447,52 @@ interface WithLoadBalancerProfile<ParentT> {
                 WithAttach<ParentT> withLoadBalancerSku(LoadBalancerSku loadBalancerSku);
             }
 
+            /**
+             * The stage of a network plugin mode definition allowing to specify the network policy.
+             *
+             * @param <ParentT> the stage of the network plugin mode definition to return to after attaching this definition
+             */
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java:454]: Doc
"The stage of a network profile definition" for all of these interfaces.

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java [hunk 3]
```diff
@@ -447,6 +447,52 @@ interface WithLoadBalancerProfile<ParentT> {
...
+                 * Specifies the network plugin mode to be used for building the Kubernetes network.
+                 *
+                 * @param networkPluginMode the network plugin mode to be used for building the Kubernetes network
+                 * @return the next stage of the definition
+                 */
+                WithAttach<ParentT> withNetworkPluginMode(NetworkPluginMode networkPluginMode);
+            }
+
+            /**
+             * The stage of a network mode definition allowing to specify the network policy.
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java:466]: to specify the network plugin mode

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java [hunk 4]
```diff
@@ -447,6 +447,52 @@ interface WithLoadBalancerProfile<ParentT> {
...
+                 * Specifies the network plugin mode to be used for building the Kubernetes network.
+                 *
+                 * @param networkMode the network mode to be used for building the Kubernetes network
+                 * @return the next stage of the definition
+                 */
+                WithAttach<ParentT> withNetworkMode(NetworkMode networkMode);
+            }
+
+            /**
+             * The stage of a network mode definition allowing to specify the network policy.
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java:481]: same problem here

**Review by weidongxu-microsoft — COMMENTED** [review]
changelog as well

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java (hunk 1 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java]: 1. `NetworkMode` and `NetworkPluginMode` cannot be specified if `NetworkPlugin` is anything other than `azure`.
2. `NetworkMode` and `NetworkPlugin` are important properties for create `aks`, so they can only be update by redeploy.

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java (hunk 4 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java]: Fixed in the new version.

File: sdk/resourcemanager/azure-resourcemanager-containerservice/CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,9 @@
 ### Features Added
 
 - Supported disabling public network access in `KubernetesCluster` via `disablePublicNetworkAccess()`, for private link feature.
+- Supported specifying network mode of Azure CNI configuration for `KubernetesCluster` during create.
+- Supported specifying network plugin mode for `KubernetesCluster` during create.
+- Supported specifying nNetwork data plane for `KubernetesCluster` during create.
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-containerservice/CHANGELOG.md:10]: typo

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**